bigquery-etl.py

```diff
@@ -0,0 +1,83 @@
+import math
+import sys
+from google.cloud import bigquery
+from fx_crash_sig.crash_processor import CrashProcessor
+
+QUERY_TEMPLATE = """
+SELECT
+  document_id, payload
+FROM
+  `moz-fx-data-shared-prod`.telemetry.crash
+WHERE
+  normalized_channel="nightly"
+  AND DATE(submission_timestamp)="{date}"
+  AND application.build_id > FORMAT_DATE("%Y%m%d", DATE_SUB(DATE "{date}", INTERVAL 1 WEEK))
```

Comment:
This query might need a bit of refining, I guess - ideally we want to generate signatures for all the crash pings, not restricted to the nightly channel, and not restricting the buildids. Assuming the airflow thing runs once a day on the previous day's submissions and no new submissions come in after the airflow task runs, we should just need the `submission_timestamp` filter here.

Reply:
I need to spend some time looking at how Tecken handled that symbolication batch on 8/20. Did it cache a lot? Did it cause problems for other parts of the service?

Short-to-medium term, I want to redo how Tecken does symbolication so it's doing it on a dedicated cluster and thus symbolication outages don't affect symbol uploads.

So my current gut feeling is that we should:

1. pick a small sample for now that Tecken can handle
2. increase the sample in the future after I make Tecken changes and/or we figure out a better way to do symbolication

I'll look into how Tecken performed today and get back to you.

Reply:
Ok, in that case maybe we should start with just nightly crashes as it's valuable to see changes in the top crashers on a daily basis. And the volume is low enough that hopefully it shouldn't be too much of a problem.

Reply:
I had a long talk about Tecken today with Brian.

Here's what we're thinking:

1. I'm in the process of redoing symbolication in Tecken. When that's done, it should work better for bigquery-etl. I'm planning to use this job to load test. I don't have an ETA--I'll probably know more next week as I flesh things out. The work is being done in this bug: https://bugzilla.mozilla.org/show_bug.cgi?id=1636210
2. This bigquery-etl job does a burst of symbolication requests which doesn't give Tecken enough time to scale up. It'd be better if somehow it could ramp up the rate of requests over 5 minutes.
3. We should keep the sample rate small for now. We can probably play with that number to see how high we can get it, if you want.